project/BuildPlugin.scala

```diff
@@ -357,6 +371,31 @@ object BuildImplementation {
       ),
     )
 
+    val gradlePluginBuildSettings: Seq[Def.Setting[_]] = List(
+      Keys.resolvers ++= List(
+        MavenRepository("Gradle releases", "https://repo.gradle.org/gradle/libs-releases-local/")
+      ),
+      Keys.libraryDependencies ++= List(
+        Dependencies.gradleCore,
+        Dependencies.gradleToolingApi,
+        Dependencies.groovy
+      ),
+      Keys.publishLocal := Keys.publishLocal.dependsOn(Keys.publishM2).value,
+      Keys.unmanagedJars.in(Compile) := unmanagedJarsWithGradleApi.value,
+      BuildKeys.fetchGradleApi := {
+        val logger = Keys.streams.value.log
+        // TODO: we may want to fetch it to a custom unmanaged lib directory under build
+        val targetDir = (Keys.baseDirectory in Compile).value / "lib"
+        GradleIntegration.fetchGradleApi(Dependencies.gradleVersion, targetDir, logger)
+      },
+    )
+
+    lazy val unmanagedJarsWithGradleApi: Def.Initialize[Task[Keys.Classpath]] = Def.taskDyn {
```

Comment:
This is a trick I did so that it `fetchGradleApi` is done exactly before it's required (this is important so that IntelliJ and other tools using the output of library dependencies of a project can see the lib jar if it's not yet downlaoded).